src/compute-client/src/controller.rs

```diff
@@ -369,27 +372,31 @@ where
     ///
     /// This method may block for an arbitrarily long time.
     ///
-    /// When the method returns, the caller should call [`ActiveComputeController::process`] with
-    /// the returned compute instance ID, to process the ready message.
+    /// When the method returns, the caller should call [`ActiveComputeController::process`] to
+    /// process the ready message.
     ///
     /// This method is cancellation safe.
-    pub async fn ready(&mut self) -> ComputeInstanceId {
+    pub async fn ready(&mut self) {
+        if self.stashed_response.is_some() {
+            // We still have a response stashed, which we are immediately ready to process.
```

Comment:
is here a `return` missing? Or maybe better an assert?

Reply:
Oof, good catch!